post/webnote-extension/index.md

```diff
@@ -0,0 +1,658 @@
+---
+title: "Build a Directus Powered WebNote Keeper Chrome Extension"
+description: "The idea is to create a Chrome extension called "WebNote Keeper" that allows users to save notes related to webpages they visit. The extension will leverage Directus as the backend CMS to store and manage these notes. When a user clicks the extension while browsing a webpage, it will automatically capture the URL of the current webpage and prompt the user to add a note. Users will be able to view, edit, and delete their notes directly from the extension's interface. The extension will integrate with Directus API for CRUD operations on notes. This project combines the use of Directus as a headless CMS and Chrome extension development skills to create a practical tool for users to organize their web research and thoughts."
+author:
+    name: "jay bharadia"
+    avatar_file_name: "my-avatar.jpg"
+---
+
+## Introduction
+
+This article will guide you through how to make a chrome extension name WebNote Keeper using `vite` and `Directus`. The extension will leverage Directus as the backend CMS to store and manage these notes.
+
+When a user clicks the extension while browsing a webpage, it will automatically capture the URL of the current webpage and prompt the user to add a `note`. Users will be able to `view`, `edit`, and `delete` their notes directly from the extension's interface. The extension will integrate with Directus API for CRUD operations on notes.
+
+## Before You Start
+
+-   A Directus project. Follow the [Quickstart guide](https://docs.directus.io/getting-started/quickstart) to create one.
+
+## Setup Directus Schema
+
+### Create Notes Collection
+
+-   `website` - new field with type `string`
+-   `note` - new field with type `string`
+
+![Notes Collection](./images/notes-collection.png)
+
+![Notes Data Entry](./images/notes-data-entry.png)
+
+### Setup Roles
+
+In order to create new note, edit note or delete note on behalf of user, we need to create separate role ( except admin ).
+We will create `customer` role for different users.
+
+-   Go to `settings` -> `Access Control` Menu
+-   Click on `Add New` Role
+-   Enter role name, select `App Access` only
+-   Save
+
+![Role access](./images/role-access-control.png)
+
+### Setup Permission
+
+-   After creating this, we need to provide `notes` collection `read`, `edit`, `add` and `delete` permission
+-   Go to `Access Control` -> Select `Customer` Role
+-   Give access to `notes` collection
+
+![Customer Role Permissions](./images/customer-role-permissions.png)
+
+Apply Filter to `notes` collection to provide data for loggedin users individually.
+
+-   Click on `eye` column -> Select `Use Custom`
+
+![Filter Notes](./images/filter-notes.png)
+
+### Directus setup is completed. Lets move to project.
```

Comment:
````suggestion
This article will guide you through building a Chrome extension using Vite and Directus. The extension will leverage Directus as the backend to store and manage data.

When a user clicks the extension while browsing a webpage, it will automatically capture the URL of the current webpage and prompt the user to add a `note`. Users will be able to `view`, `edit`, and `delete` their notes directly from the extension.

Before you start, you will need a Directus project. Follow the [Quickstart guide](https://docs.directus.io/getting-started/quickstart) to create one if needed.

This tutorial will not cover styling the extension, but light styling has been applied to the screenshots shown.

## Set Up Your Directus Project

### Create Notes Collection

Create a new collection called `notes` with all optional fields enabled. Create the following additional fields:

-   `website` - new field with type `string`
-   `note` - new field with type `string`

![Notes Collection](./images/notes-collection.png)

### Configure Permissions

Create a `customer` role for new users. In the access control settings for the role, provide

-   Go to `Access Control` -> Select `Customer` Role.
-   Set a custom permission for all operations in the `notes` collection: `user_created equals $CURRENT_USER`.

![Filter Notes](./images/filter-notes.png)
````

You have to apply the custom permissions to all operations, else a user can edit everyone's notes, delete everyone's notes, etc.